services/QuillLMS/engines/comprehension/app/controllers/comprehension/rules_controller.rb

```diff
@@ -45,6 +45,23 @@ def destroy
       head :no_content
     end
 
+    def update_rule_order
+      ordered_rules = ordered_rules_params[:ordered_rule_ids].map.with_index do |id, index|
+        rule = Comprehension::Rule.find_by_id(id)
```

Comment:
_**If**_ you wanted to include errors for the nil rules you could utilize the [Null object pattern](https://thoughtbot.com/blog/rails-refactoring-example-introduce-null-object).

```
module Comprehension
  class NullRule
    attr_reader :errors
    attr_accessor :suborder

    def initialize(errors: [])
      @errors = errors
    end

    def valid?
      false
    end
  end
end
```

Which would allow for the pretzel colon shorthand:

```
def update_rule_order
  ordered_rules = ordered_rules_params[:ordered_rule_ids].map.with_index do |id, index|
    rule = Comprehension::Rule.find_by_id(id) || Comprehension::NullRule.new(errors: "rule #{id} was not found")
    rule.suborder = index
    rule
  end

  if ordered_rules.all?(&:valid?)
    ordered_rules.each(&:save!)
    render json: { status: 200 }
  else
    render json: { error_messages: ordered_rules.map(&:errors).join('; ') }, status: :unprocessable_entity
  end
end
```